src/main/java/Game.java

```diff
@@ -1,65 +1,185 @@
+				break;
+			case "score":
+				result = "Your score is currently: " + player.getScore();
+				break;
+			case "text":
+			case "deliver":
+			case "shout":
+			case "broadcast":
+				deliverMessage(command, player.getName());
+				result = "message delivered ";
+				break;
+			case "lift":
+			case "open":
+				result = handleContainer(command.getItems().get(0), player, action);
 				break;
 			case "help":
 			case "h":
-				giveHelpInstructions(player);
+				result = getHelpInstructions(player);
 				break;
+
 			default:
-				player.output("I do not understand " + action);
+				result = ("I do not understand " + action);
+		}
+
+		//Movement shortcut
+		if (isDirection(action)) { result = handleMove(action, player); }
+		isBusy = false; //release lock.
+		return result;
+	}
+
+	private static Item hasItem(List<Item> contents, String item) {
+		Item found = null;
+		for(Item i : contents) {
+			if(i.getName().equals(item)) {
+				found = i;
+			}
+		}
+		return found;
+	}
+
+	private static String giveItem(List<String> items, Player player) {
+		try {
+			String giveWhat = items.get(0);
+			String toWhat = items.get(2);
+
+			if (!items.get(1).equals("to")) {
+				return "I do not understand what you mean";
+			}
+			if (hasItem(player.getBackpack(), giveWhat) == null) {
+				return "You do not have a " + giveWhat + " in your backpack.";
+			}
+			if (!neutralizeObstacles(player.position(), giveWhat, toWhat)) {
+				return "There is no " + toWhat + " in this room.";
+			}
+		} catch (NullPointerException e) {
+			return "I do not understand what you mean";
+		}
+		return "You have neutralized the obstacle, cool! \n" + map.getDescription(player.position());
+	}
+
+	private static boolean neutralizeObstacles(Coordinate coord, String toNeutralize, String who) {
+		Area area = map.getArea(coord);
+		if (area.getObstacle() != null && area.getObstacle().getName().equals(who) &&
+				area.getObstacle().getHowToNeutralize().equals(toNeutralize))
+		{
+			area.setObstacle(null);
+			return true;
 		}
+		return false;
 	}
 
-	private static void handleMove(String direction, Player player) {
-		Coordinate newPos = null;
+	private static String handleItem(String item, Player player, List<Item> contents, String action) {
+		Item toRemove = hasItem(contents, item);
+		String result;
+		if(toRemove == null) {
+			if(action.equals("drop")) {
+				result = "You cannot drop what you do not have, there is not " + item + " in your backpack.";
+			} else {
+				result = "I don't see " + item + " anywhere in here.";
+			}
+		} else {
+			if(action.equals("drop")) {
+				result = dropItem(item, toRemove, player);
+			} else {
+				result = takeItem(item, toRemove, player);
+			}
+		}
+		return result;
+	}
+
+	private static String dropItem(String item, Item removeItem, Player player ){
+		map.getArea(player.position()).addItem(removeItem);
+		player.removeItem(removeItem);
+		return "Your backpack is so much lighter when there is no " + item + " in it.";
```

Comment:
Throughout your code, there is a good mix of hard coding of dialogue with the inclusion of for instance items removed in this case that make the system interesting and unique to the individual user.